lib/Parser/Parser.cpp

```diff
@@ -163,13 +168,75 @@ Token Lexer::getNextToken(std::string &ErrStr) {
     }
   }
 
-  if ((*Begin >= 'a' && *Begin <= 'z') || (*Begin == '.')) {
+  if ((*Begin >= 'a' && *Begin <= 'z') || (*Begin == '.') ||
+      (*Begin >= 'A' && *Begin <= 'Z')) {
     const char *TokenBegin = Begin;
     do {
       ++Begin;
     } while (Begin != End && ((*Begin >= 'a' && *Begin <= 'z') ||
-             (*Begin == '.')));
-    return Token{Token::Ident, TokenBegin, size_t(Begin - TokenBegin), APInt()};
+             (*Begin == '.') || (*Begin >= 'A' && *Begin <= 'Z')));
+    std::string DataFlowFact = StringRef(TokenBegin, Begin - TokenBegin);
+    const char *PatternBegin;
+    Token T;
+    if (DataFlowFact == "knownBits") {
```

Comment:
it seems like there should also be a way to properly lex (knownBits=01x) as five tokens, without needing to backtrack.

What would happen if we modified UntypedInt so that it could also contain "x" characters?

Reply:
or just leave this as a disgusting hack for now but make the other changes and I can try to fix it later